Replace the `BookRepository` entity with one that uses `JpaRepository`.

Oleh8 opened the pull request:
Add Liquibase support.
Change the `spring.jpa.hibernate.ddl-auto` property value to `validate`. Implement the soft delete concept in the `Book` entity. Add missing endpoints and methods to `BookService`.

Oleh8 committed:
Replace the `BookRepository` entity with one that uses `JpaRepository`.
Add Liquibase support.
Change the `spring.jpa.hibernate.ddl-auto` property value to `validate`.
Implement the soft delete concept in the `Book` entity.
Add missing endpoints and methods to `BookService`.

Oleh8 committed:
Replace the `BookRepository` entity with one that uses `JpaRepository`.
Add Liquibase support.
Change the `spring.jpa.hibernate.ddl-auto` property value to `validate`.
Implement the soft delete concept in the `Book` entity.
Add missing endpoints and methods to `BookService`.

liliia-ponomarenko commented on src/main/java/bstore/bookstore/service/BookServiceImpl.java:
```diff
@@ -7,15 +7,20 @@
     private final BookRepository bookRepository;
     private final BookMapper bookMapper;
 
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository, BookMapper bookMapper) {
+        this.bookRepository = bookRepository;
+        this.bookMapper = bookMapper;
+    }
+
```
Let’s  use RequiredArgsConstructor

liliia-ponomarenko commented on src/main/java/bstore/bookstore/service/BookServiceImpl.java:
```diff
@@ -35,4 +40,22 @@ public List<BookDto> findAll() {
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
+        book.setTitle(requestDto.getTitle());
+        book.setAuthor(requestDto.getAuthor());
+        book.setIsbn(requestDto.getIsbn());
+        book.setPrice(requestDto.getPrice());
+        book.setDescription(requestDto.getDescription());
+        book.setCoverImage(requestDto.getCoverImage());
```
use mapper for it. Create method `Book updateBook(Book book, CreateBookRequestDto requestDto)`

liliia-ponomarenko commented on src/main/resources/db/changelog/changes/01-create-books-table.yaml:
```diff
@@ -0,0 +1,47 @@
+                  name: description
+                  type: varchar(255)
+              - column:
+                  name: coverImage
```
better use snake_case format

liliia-ponomarenko commented on src/main/resources/db/changelog/changes/01-create-books-table.yaml:
```diff
@@ -0,0 +1,47 @@
+                  name: isDeleted
+                  type: tinyint(1)
+                  constraints:
+                    nullable: false
```
leave one empty line at the end of the file (check other files)

liliia-ponomarenko reviewed (changes requested):
Good job! Let’s improve your solution!

Oleh8 committed:
Replace the `BookRepository` entity with one that uses `JpaRepository`.
Add Liquibase support.
Change the `spring.jpa.hibernate.ddl-auto` property value to `validate`.
Implement the soft delete concept in the `Book` entity.
Add missing endpoints and methods to `BookService`.

Elena-Bruyako commented on src/main/java/bstore/bookstore/mapper/BookMapper.java:
```diff
@@ -5,11 +5,14 @@
 
     Book toModel(CreateBookRequestDto requestDto);
+
+    Book updateBook(@MappingTarget Book book, CreateBookRequestDto requestDto);
```
```suggestion
    void updateBook(@MappingTarget Book book, CreateBookRequestDto requestDto);
```

Elena-Bruyako commented on src/main/java/bstore/bookstore/model/Book.java:
```diff
@@ -7,17 +7,15 @@
 
 @Entity
-@Table(name = "books")
 @Data
```
Don't use `@Data` for entities
```suggestion
@Getter
@Setter
```

Elena-Bruyako commented on src/main/java/bstore/bookstore/service/BookServiceImpl.java:
```diff
@@ -35,4 +35,16 @@ public List<BookDto> findAll() {
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
+        return bookMapper.toDto(bookRepository.save(bookMapper.updateBook(book, requestDto)));
```
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));
        bookMapper.updateBook(book, requestDto);
        return bookMapper.toDto(bookRepository.save(book));
```

Elena-Bruyako reviewed (changes requested):
see comments

Oleh8 committed:
Replace the `BookRepository` entity with one that uses `JpaRepository`.
Add Liquibase support.
Change the `spring.jpa.hibernate.ddl-auto` property value to `validate`.
Implement the soft delete concept in the `Book` entity.
Add missing endpoints and methods to `BookService`.
